### DISCUSSION TITLE
auto learning rate finder problem

### DISCUSSION THREAD

**Discussion by cmlakhan** [Lightning-AI/pytorch-lightning#13120] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
In this example for auto learning rate finder it seems learning rate is a parameter for model initialization yet it not specified when the model is called.  When I do something similar I get an error saying I have not specified the variable learning_rate.  Can someone explain what is happening here?

```
class LitModel(LightningModule):
    def __init__(self, learning_rate):
        super().__init__()
        self.learning_rate = learning_rate
        self.model = Model(...)

    def configure_optimizers(self):
        return Adam(self.parameters(), lr=(self.lr or self.learning_rate))


model = LitModel()

# finds learning rate automatically
# sets hparams.lr or hparams.learning_rate to that learning rate
trainer = Trainer(auto_lr_find=True)

trainer.tune(model)
```

**Comment by akihironitta** [comment]
@cmlakhan Can you try giving an initial value to it and see if it works? Also, would you be interested in fixing the example in our docs?
